daemon/health.go

```diff
@@ -231,8 +231,11 @@ func handleProbeResult(d *Daemon, c *container.Container, result *types.Healthch
 		// Else we're starting or healthy. Stay in that state.
 	}
 
-	// replicate Health status changes
-	if err := c.CheckpointTo(d.containersReplica); err != nil {
+	// Replicate Health status changes to the API, skipping persistent storage
+	// to avoid unnecessary disk writes. The health state is only best-effort
+	// persisted across of the daemon. It will get written to disk on the next
+	// checkpoint, such as when the container state changes.
```

Comment:
Do long running containers (that are most likely to have the healthchecks) have enough opportunities for state change? Maybe we could also do the checkpoint to disk when the daemon is shutting down cleanly and live restore is enabled? Unless it's already a thing, but taking a quick glance at the code I didn't find anything that would do that directly.

Reply:
Read the commit messages again. I explain why there is no need to checkpoint on shutdown when live-restore is enabled.

Reply:
Oh right, I missed it somehow.